### ISSUE TITLE
Solana Support

### ISSUE THREAD

**Issue by adrien-be** [rotki/rotki#2394]
## Abstract

As Solana growing bigger I see the potential for Rotki to capture new userbase and it will help to adapt for projects based on Solana.

## Motivation

Solana blockchain is now live and kicking. The whole [Solana ecosystem](https://solana.com/ecosystem) is growing fast.

For those unfamiliar with Solana: 

- it implements Optimistic Concurrency Control and route towards that theoretical limit of 710K TPS —  https://docs.solana.com/introduction#why-solana
- does **not** use sharding
- 50K tps in testnet
- 400+ Validators on Mainnet https://solanabeach.io/validators
- Explorer — https://explorer.solana.com/
- Team — https://www.linkedin.com/company/solanalabs/people/

## Specification

Just created this as coordination issue

Solana docs on https://docs.solana.com/

## Grant

You can possibly get a grant from Solana Foundation: https://solana.com/grants

Solana has been very supportive with its community, maybe that's another avenue for funding this effort: https://solana.com/collective

DeFi hackathon — maybe a bit of a stretch, but I preferred mentioning it should that be another avenue for funding this effort https://solana.com/defi

**Comment by lambda-0x** [comment]
+1

**Comment by helloeddy** [comment]
+1

**Comment by faraggi** [comment]
Anybody actively working on this?

**Comment by LefterisJP** [comment]
> Anybody actively working on this?

no this is not worked on at the moment

**Comment by phenomax** [comment]
+1

**Comment by faraggi** [comment]
In that case I might take it on- it'll might help me better understand the solana technicals. 
I'll update this to confirm starting work, and further updates.

**Comment by yabirgb** [comment]
> In that case I might take it on- it'll might help me better understand the solana technicals. I'll update this to confirm starting work, and further updates.

Thanks for taking this. Here you have the PR introducing support for avalanche. Since solana is not evm compatible afaik details might be different but at least you get an idea of what changes where needed and what code was introduced in case it can be useful. 

https://github.com/rotki/rotki/pull/3242

**Comment by scriptless** [comment]
Premium user here. I would very much appreciate this being implemented soon, thanks!

**Comment by adigupta13** [comment]
+1

**Comment by faraggi** [comment]
Update: I've spoken to the Solana grants team last week, and they didn't accept giving this project a grant, so I will be taking a step aside and giving anyone else the chance to move forward with it.

**Comment by phenomax** [comment]
https://github.com/hodgerpodger/staketaxcsv/tree/main/src/sol could be a good starting point, as they are grabbing and parsing tx data via rpc.

**Comment by xAltsuki** [comment]
+1

**Comment by zehytien** [comment]
Would of paid for premium in a second if Solana was in that drop down.

**Comment by helloeddy** [comment]
+1

**Comment by kanylbullen** [comment]
I would love to see this worked on. Have you applied for a grant at the Solana foundation?

**Comment by faraggi** [comment]
I did and they refused.
But it was years ago and it might go differently today.

You may want to try again.

**Comment by damian-w** [comment]
Is this being actively worked on?

**Comment by yabirgb** [comment]
Hello @damian-w . Not at the moment

**Comment by beautyfree** [comment]
where i can vote for prioritization of this feature?

**Comment by maximecharriere** [comment]
+1. Would love to track my SOL tokens on Rotki.  
Why do we need a grant from Solana Foundation ? I know nothing about Rotki, but isn't it possible to simply use https://explorer.solana.com/ API to track transactions history?

**Comment by LefterisJP** [comment]
Software development costs money @faraggi, and our main source of income is from grants

In any case I am happy to say this has been scheduled to be completed by end of this year. We are going to work on it.

**Comment by maximecharriere** [comment]
I totally agree that software dev cost money and developers needs to be payed!
Either by grants or by Rotki premium users.

Happy to ear it's scheduled 😍. Solana is still the 2nd biggest ecosystem behind Ethereum (I think).

**Comment by kanylbullen** [comment]
Income from premium subscriptions should be able to increase if the product is more complete with all major chains and exchanges? 

Speaking for myself, the main issue  holding me back from paying for premium is that I still need to pay for coinstats premium to be able to have one place with a complete overview of portfolios, but I'd much rather be fully self-hosted.

As a sidenote I think the premium price model could improve to attract more subscribers. More differentiated prices for longer subscription times, and a lifetime option maybe?

**Comment by LefterisJP** [comment]
I love the engagement guys! But let's keep the issue on topic especially of the implementation now that we are going to schedule this.

If you want to discuss anything else feel free to pop up in discord: https://discord.rotki.com/

**Comment by SaintelDaily** [comment]
Rocket to show your support for a  **lifetime option!**

**Comment by Fabelwesen** [comment]
+1

**Comment by nicholasyoder** [comment]
Will be included in v1.41
